tests/drivers/timer/nrf_grtc_timer/src/main.c

```diff
@@ -37,4 +49,79 @@ ZTEST(nrf_grtc_timer, test_get_ticks)
 		     "Unexpected result %" PRId64 " (expected: %" PRId64 ")", ticks, exp_ticks);
 }
 
+ZTEST(nrf_grtc_timer, test_timer_count_in_compare_mode)
+{
+	int err;
+	int key;
+	uint64_t test_ticks = 0;
+	char user_data[] = "test_timer_count_in_compare_mode\n";
+	int32_t channel = z_nrf_grtc_timer_chan_alloc();
+
+	TC_PRINT("Allocated GRTC channel %d\n", channel);
+	if (channel < 0) {
+		TC_PRINT("Failed to allocate GRTC channel, chan=%d\n", channel);
+		ztest_test_fail();
+	}
+
+	compare_isr_call_counter = 0;
+	key = z_nrf_grtc_timer_compare_int_lock(channel);
+	test_ticks = z_nrf_grtc_timer_get_ticks(K_MSEC(TIMER_COUNT_TIME_MS));
+	err = z_nrf_grtc_timer_set(channel, test_ticks, timer_compare_interrupt_handler,
+				   (void *)user_data);
+	z_nrf_grtc_timer_compare_int_unlock(channel, key);
+
+	zassert_true(K_TIMEOUT_EQ(K_TICKS(z_nrf_grtc_timer_compare_read(channel)),
+				  K_TICKS(test_ticks)),
+		     "Compare register set failed");
+
+	zassert_equal(err, 0, "Unexpected error raised when setting timer, err: %d", err);
+
+	k_sleep(K_MSEC(WAIT_FOR_TIMER_EVENT_TIME_MS));
+
+	TC_PRINT("Compare event generated ?: %d\n", z_nrf_grtc_timer_compare_evt_check(channel));
+	TC_PRINT("Compare event register address: %X\n",
+		 z_nrf_grtc_timer_compare_evt_address_get(channel));
+
+	zassert_equal(compare_isr_call_counter, 1);
+
+	z_nrf_grtc_timer_chan_free(channel);
+}
+
+ZTEST(nrf_grtc_timer, test_timer_abort_in_compare_mode)
+{
+	int err;
+	int key;
+	uint64_t test_ticks = 0;
+	char user_data[] = "test_timer_abort_in_compare_mode\n";
+	int32_t channel = z_nrf_grtc_timer_chan_alloc();
+
+	TC_PRINT("Allocated GRTC channel %d\n", channel);
+	if (channel < 0) {
+		TC_PRINT("Failed to allocate GRTC channel, chan=%d\n", channel);
+		ztest_test_fail();
+	}
+
+	compare_isr_call_counter = 0;
+	key = z_nrf_grtc_timer_compare_int_lock(channel);
+	z_nrf_grtc_timer_compare_int_unlock(channel, key);
```

Comment:
What is the point in doing this?

Reply:
How do I enable the interrupt otherwise, the API documentation states that 'key' is needed.

```
/** @brief Safely enable compare event interrupt.
 *
 * Event interrupt is conditionally enabled based on @p key.
 *
 * @param chan Channel ID.
 *
 * @param key Key returned by z_nrf_grtc_timer_compare_int_lock().
 */
void z_nrf_grtc_timer_compare_int_unlock(int32_t chan, bool key);

/** @brief Read compare register value.
 *
 * @param chan Channel ID.
 *
 * @retval >=0 Positive is a Value set in the compare register
 * @retval -EAGAIN if compare for given channel is not set.
 * @retval -EPERM if either channel is unavailable or SYSCOUNTER is not running.
```

Reply:
> How do I enable the interrupt otherwise, the API documentation states that 'key' is needed.

You don't need to enable the interrupts. The driver does it when you set up the compare channel:
https://github.com/zephyrproject-rtos/zephyr/blob/d0be2010c31fed8ade258dae0417efdc02eca235/drivers/timer/nrf_grtc_timer.c#L288-L305
For the same reason the calls to `_int_lock()` and `_int_unlock()` in `test_timer_count_in_compare_mode` are redundant.